### DISCUSSION TITLE
distributive law option?

### DISCUSSION THREAD

**Discussion by davies-w** [josdejong/mathjs#3240] [Q&A]
I'm trying to figure out if its possible to switch between formulations. I've tried using Simplify, and it doesn't seem to do much at all for this kind of algebraic manipulation.

(a + b) * (c +d)   <=> (a *c) + (b*c) + (a*d) + (b*d)

I'm trying to do a little bit of poprositional algebra trivial query rewriting, switching between AND and ORs of positive literals, but it's the same Distributive Law as in regular algebra, so I thought this library might be an off the shelf solution.

**Comment by davies-w** [comment]
woot!!
 
mathjs.rationalize("(a+b)*(c+d)*(e+f)").toString()
'e * a * c + f * a * c + e * a * d + f * a * d + e * b * c + f * b * c + e * b * d + f * b * d'

**Reply by davies-w** [reply]
However, this doesn't go the other way around 

'e * a * c + f * a * c + e * a * d + f * a * d + e * b * c + f * b * c + e * b * d + f * b * d' => (a+b)(c+d)(e+f)"

**Comment by josdejong** [comment]
I think you can utilize `simplify` for this when you provide your own `rules`, just applying the normalization that you're looking for.

Docs: https://mathjs.org/docs/reference/functions/simplify.html
